File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        int m=INT_MAX,c=0,f=0;
        long long s=0;
        for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[0].size();j++){
                if(matrix[i][j]<0){
                    c++;
                    s+=abs(matrix[i][j]);
                }
                else if(matrix[i][j]==0) f=1;
                else s+=matrix[i][j];
                m=min(m,abs(matrix[i][j]));
            }
        }
        if(c==0 || c%2==0 || f==1) return s;
        return s-m-m;
    }
};

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        int s=0,min=Integer.MAX_VALUE,c=0,f=0;
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j]<0){
                    c++;
                    s+=Math.abs(matrix[i][j]);
                }
                else if(matrix[i][j]==0) f=1;
                else s+=matrix[i][j];
                min=Math.min(min,Math.abs(matrix[i][j]));
            }
        }
        if(c==0 || c%2==0 || f==1) return s;
        return s-min-min;
    }
}